apps/electionguard-cli/Generate/GenerateCommand.cs

```diff
@@ -0,0 +1,110 @@
+﻿using ElectionGuard.Encryption.Utils.Generators;
+using ElectionGuard.Extensions;
+using Newtonsoft.Json;
+
+namespace ElectionGuard.CLI.Generate
+{
+    internal class GenerateCommand
+    {
+        public static Task Execute(GenerateOptions options)
+        {
+            try
+            {
+                var command = new GenerateCommand();
+                return command.ExecuteInternal(options);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                throw;
+            }
+        }
+
+        private async Task ExecuteInternal(GenerateOptions options)
+        {
+            options.Validate();
+
+            PlainTally? tally = null;
+            var context = Path.Combine(options.WorkingDir, "context.json");
+            var manifest = Path.Combine(options.WorkingDir, "manifest.json");
+            using var internalManifest = GetInternalManifest(manifest);
+            using var encryptionMediator = GetEncryptionMediator(context, manifest);
+
+            var plaintextBallots = BallotGenerator.GetFakeBallots(
+                internalManifest, Random.Shared, options.BallotCount);
+
+            var plaintextPath = Path.Combine(options.WorkingDir, "plaintext");
+            var encryptedPath = Path.Combine(options.WorkingDir, "encrypted");
+
+            foreach (var ballot in plaintextBallots)

```

Comment:
This would be useful to try a parallel foreach

Reply:
We did try this together and we will need to figure out how to parallelize this in the future.